pkg/rsyslog/init.sh

```diff
@@ -1,20 +1,8 @@
 #!/bin/sh
 
-RSYSLOG_WORK_DIR=/persist/rsyslog
-OLD_DIR=/persist/syslog
-# We need to clean up old state it seems. Use $OLD_DIR as the hint
-if [ -d "$OLD_DIR" ]; then
-    echo "Moving old $OLD_DIR and $RSYSLOG_WORK_DIR out of the way"
-    mv "$OLD_DIR" "$OLD_DIR".old
-    mv "$RSYSLOG_WORK_DIR" "$RSYSLOG_WORK_DIR".old
-fi
-if [ ! -d "$RSYSLOG_WORK_DIR" ]; then
-  mkdir -p $RSYSLOG_WORK_DIR
-  chmod 644 $RSYSLOG_WORK_DIR
-fi
-
-IMGP=$(cat /run/eve.id 2>/dev/null)
-IMGP=${IMGP:-IMGX} /usr/sbin/rsyslogd
+./start-rsyslogd.sh
 
 # XXX this can hang forever?? And we don't have a way to recover
 # waitforsyslog
+
+./monitor-rsyslog.sh
```

Comment:
If rsyslogd doesn't start for some reason, this will result in a hardware watchdog (if that's enabled on the platform) or a hang forever (if no hardware watchdog).
Furthermore, if the reason for the failure is the same post the hardware watchdog, then it will repeat forever. 
So we can't do this (for the same reason that the commented out waitforsyslog on line 6 didn't work.

Reply:
This loop does not stop other containers from starting. If rsyslogd does not start, watchdog in pillar container will eventually reboot the device.

Reply:
Yeah -- I think @gkodali-zededa gets it exactly right @nordmark 

Now, here's a subtle point -- the criteria for "giving up" now resides at the monitor-rsyslog.sh level -- that's the PID I want watchdog to monitor -- not rsyslog itself.

That way -- we can decide when "enough is enough" in trying to bring rsyslog back from the dead

Reply:
Ah got it. So we need a pidfile from monitor-rsyslog.sh in /run so watchdog can look at it.

Reply:
@rvs @eriknordmark Sure, I'll make watchdog monitor this script PID instead of rsyslogd pid.